src/core/algorithms/cfd/cfd_verifier/cfd_verifier.cpp

```diff
@@ -0,0 +1,94 @@
+#include "cfd_verifier.h"
+
+#include <easylogging++.h>
+
+#include "cfd/model/cfd_relation_data.h"
+#include "cfd/util/cfd_output_util.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+#include "util/timed_invoke.h"
+
+namespace algos::cfd_verifier {
+
+CFDVerifier::CFDVerifier() : Algorithm({}) {
+    RegisterOptions();
+    MakeOptionsAvailable({config::kTableOpt.GetName()});
+}
+
+void CFDVerifier::RegisterOptions() {
+    DESBORDANTE_OPTION_USING;
+    RegisterOption(config::kTableOpt(&input_table_));
+    RegisterOption(Option{&string_rule_left_, kCFDRuleLeft, kDCFDRuleLeft,
+                          std::vector<std::pair<std::string, std::string>>{}});
+    RegisterOption(Option{&string_rule_right_, kCFDRuleRight, kDCFDRuleRight,
+                          std::pair<std::string, std::string>{}});
+    RegisterOption(Option{&minconf_, kMinimumConfidence, kDMinimumConfidence, 0.0});
+    RegisterOption(Option{&minsup_, kMinimumSupport, kDMinimumSupport, 0});
+}
+
+void CFDVerifier::LoadDataInternal() {
+    relation_ = cfd::CFDRelationData::CreateFrom(*input_table_);
+
+    if (relation_->GetColumnData().empty()) {
+        throw std::runtime_error("Got an empty dataset: CFD verifying is meaningless.");
+    }
+}
+
+void CFDVerifier::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kCFDRuleLeft, kCFDRuleRight, kMinimumSupport, kMinimumConfidence});
+}
+
+unsigned long long CFDVerifier::ExecuteInternal() {
+    auto get_attr_id = [this](std::string const& attr_name) -> cfd::AttributeIndex {
+        cfd::AttributeIndex attr_id = relation_->GetAttr(attr_name);
+        if (attr_id == -1) {
+            throw config::ConfigurationError("Attribute not found: " + attr_name);
+        }
+        return attr_id;
+    };
+
+    auto extract_item_ids =
+            [this, &get_attr_id](std::vector<std::pair<std::string, std::string>> const& rule_part)
+            -> cfd::Itemset {
+        cfd::Itemset item_ids;
+        for (auto const& [attr_name, item_name] : rule_part) {
+            cfd::AttributeIndex attr_id = get_attr_id(attr_name);
+
+            if (item_name != "_") {
+                cfd::Item item_id = relation_->GetItem(attr_id, item_name);
+                if (item_id == -1) {
+                    throw config::ConfigurationError("Item not found in item universe: " +
+                                                     item_name);
+                }
+                item_ids.push_back(item_id);
+            } else {
+                item_ids.push_back(-1 - attr_id);
```

Comment:
`-1 - attr_id` seems strange. I would like to see some minor comment with explanation

Reply:
Address this, please